### PR TITLE
Refractor passport social authentication

### PR DESCRIPTION
**Pull request by ademesojosiah**

### CODE REVIEW

**Comment by ademesojosiah** [comment]
@anslemkelechi  don't merge yet

**Comment by ademesojosiah** [comment]
you can merge now,
you can review and ask questions , I will be sure to answer

File: controllers/postController.js [hunk 1]
```diff
@@ -21,9 +21,11 @@ const createPost = async (req,res) =>{
     //     await cloudinary.uploader.upload(file.path,{folder:"media_url"}).secure_url
     // })
 
-    const filePath = req.file.path
-    const cloudinaryPath = await cloudinary.uploader.upload(filePath,{folder:"media_url"})
-    body.media_url = cloudinaryPath.secure_url
+    const filePath = req.file?.path
```
Comment (KorkuTegbe) [inline controllers/postController.js]: line 24. I'm guessing the '?' is a typo. Right?

Comment (ademesojosiah) [reply controllers/postController.js]: No it’s not , it’s called optional chaining
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining
you can read about it here

Comment (KorkuTegbe) [reply controllers/postController.js]: oh okay, thanks

**Comment by ademesojosiah** [comment]
@anslemkelechi good to merge

**Pull request merged**